### PR TITLE
Recycle model resources on reset.

### PR DESCRIPTION
**Pull request by 1chandu**
When resetting Model on layer, call Model.delete() to trigger resource recycle.
Fixes performance regression issue seen in : #1126 

This triggers Program object recycling in luma.gl (https://github.com/uber/luma.gl/pull/352)

Verified with examples, and layer-browser (change props, toggling `fp64` multiple times).

### CODE REVIEW

**Comment by georgios-uber** [comment]
Seems good to me but I don't think I have good knowledge of the rest of the code to approve.

File: src/core/lib/layer.js [hunk 1]
```diff
@@ -578,6 +579,17 @@ export default class Layer {
     }
   }
 
+  _deleteResources(newState) {
```
Comment (ibgreen) [inline src/core/lib/layer.js]: You may want to also check `state,models`

Don't we want to do this in `finalizeLayer` as well?

**Comment by 1chandu** [comment]
`forceDelete` is set only from `finalizeLayer`, the intention is to keep all `delete` logic in one place instead of spreading it into multiple places. Lets discuss this offline to understand your concerns.

**Comment by 1chandu** [comment]
Alternate solution : https://github.com/uber/deck.gl/pull/1147

**Comment by shaojingli** [comment]
@1chandu Are we plan to abandon this PR in favor of #1147?

**Comment by 1chandu** [comment]
Yes, closing.

**Pull request closed without merging**